mathics/core/parser/README.md

```diff
@@ -166,81 +174,65 @@ This is simple to handle with precedence climbing:
 
 ```python
 def p_Integral(self, token):
-    `expr <- Integral expr1 DifferentialD expr2
+    `expr ::= Integral expr1 DifferentialD expr2
     self.consume()                          # consume 'Integral'
 
-    inner_prec = all_ops['Sum'] + 1         # lookup inner
-    outer_prec = all_ops['Power'] - 1       # and outer prec
+    inner_prec = all_operators['Sum'] + 1   # lookup inner
+    outer_prec = all_operators['Power'] - 1 # and outer prec
 
     expr1 = self.parse_exp(inner_prec)      # consume expr1
     self.expect('DifferentialD')            # consume 'DifferentialD'
     expr2 = self.parse_exp(outer_prec)      # consume expr2
     return Node('Integrate', expr1, expr2)
 ```
 
-## Mathics implementation
+## Mathics3 implementation
 
 ### Precedence Climbing
-All the Mathics operators are specified in `mathics/core/parser/operators.py`.
-The precedence climbing algorithm is implemented in
-`mathics/core/parser/parser.py`. All the special cases are implemented as
-additional rules.
+All the Mathics operators are specified in `mathics/core/parser/operators.py`. The precedence climbing algorithm is implemented in
+`mathics/core/parser/parser.py`. All the special cases are implemented as additional rules.
 
 #### P Rules
 
 Methods beginning with `p_TAG` declare what to do when the `TAG` token is
 encountered at the beginning of an expression. For example, the first token
-after an open parenthesis. These rules define all the atomics `E <- A`, prefix
-operators, `E <- PREFIX E` and also brackets `E <- ( E )`.
+after an open parenthesis. These rules define all the atomics `E ::= A`, prefix
+operators, `E ::= PREFIX E` and also brackets `E ::= ( E )`.
 
 #### E Rules
 
 Methods beginning with `e_TAG` are called when one expression is already
 present and we encounter the `TAG` token. This covers binary operators
-`E <- E BINARY E`, postfix operators `E <- E POSTFIX`, ternary operators,
-`E <- E TERNARY1 E TERNARY2 E`, and functions `E <- E [ SEQUENCE ]`.
+`E ::= E BINARY E`, postfix operators `E ::= E POSTFIX`, ternary operators,
+`E ::= E TERNARY1 E TERNARY2 E`, and functions `E ::= E [ SEQUENCE ]`.
 
 #### B Rules
 
 Methods beginning with `b_TAG` are used for parsing boxes and can be ignored
 on first reading of the parser code.
 
 #### Backtracking
-Most of the Wolfram language can be parsed with precedence climbing but there
-are a few special language features that require something more. The `Span`
-operator is one example. Both `a ;; b` and `a ;;` are valid syntax, the latter
-is equivalent to `a ;; All`.
+Most of the Wolfram language can be parsed with precedence climbing but there are a few special language features that require something more. The `Span` operator is one example. Both `a ;; b` and `a ;;` are valid syntax, the latter is equivalent to `a ;; All`.
 
 Consider the example: `a;;!b`. There are four options for parsing this:
  - `Span[a, Not[b]]`
  - `Factorial[Span[a, Null]]`
```

Comment:
`Times[Factorial[Span[a,Null]], b`  ?

Reply:
When two expressions butt up against each other there is an implicit multiplication. For example `(5)3, or 3x. 

The interpretation of `Times[Factorial[Span[a,All]], b` is `(a;;!)b`.

So "All", not Null". 

I've gone over this example though in order to try to clarify it. Also, it looks like Mathics3 isn't parsing this correctly.